### DISCUSSION TITLE
Advice on Automating Multiple Checks for Updates

### DISCUSSION THREAD

**Discussion by Pamoho** [community/community#24668] [Apps, API and Webhooks]
<p>OK so there is probably something simple I’ve missed!</p>
<p>What I am trying to do is write a scheduled task (e.g. weekly) that interrogates multiple GitHub repositories to see if components our code uses have recent updates.</p>
<p>Components like Jedi Component Library and Pavkam’s TZdb Component (plus sundry others).</p>
<p>What I’ve tried so far is to open links in my App one after the other and compare Buffer Lengths / Checksums etc. with stored values.  A difference flags a change, right?</p>
<p>Well it does on other sources of Components (with probably more primitive websites), but I tend to get a flag for “change” every visit to a GitHub link, for example: <a href="https://github.com/pavkam/tzdb/releases" rel="noopener nofollow ugc">https://github.com/pavkam/tzdb/releases</a></p>
<p>It is pretty obvious why when you look at the page, but I can’t see a better way to do it.</p>
<p>Suggestions?</p>
<p>Cheers, Paul</p>
<p>PS For Alex’s software I was relying on his Blog at <a href="https://pavkam.dev/" rel="noopener nofollow ugc">https://pavkam.dev/</a> to do this in the past (worked really well), but I’ve “standardised” on linking to GitHub recently.</p>
<p>P</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="Pamoho" data-post="1" data-topic="220726">
<div class="title">
<div class="quote-controls"></div>
 Pamoho:</div>
<blockquote>
<p>What I’ve tried so far is to open links in my App one after the other and compare Buffer Lengths / Checksums etc. with stored values. A difference flags a change, right?</p>
</blockquote>
</aside>
<p>That depends on whether the thing you download is meant to only change if there is a new release. If you download the releases page for a GitHub repository, there are a lot of other things that might change.</p>
<p>Using the REST API to get the latest release should work much better, and you don’t even need the checksum: <a href="https://docs.github.com/en/rest/reference/releases#get-the-latest-release" class="inline-onebox">Releases - GitHub Docs</a></p>

**Comment by Pamoho** [comment]
<p>Thanks for the prompt response!  This looks promising, I’ll investigate further in a day or so.  It should be quite straight-forward to change my code to use the API etc. so I should get results quite soon…</p>
<p>Cheers, Paul</p>

**Comment by Pamoho** [comment]
<ul>
<li>Update *</li>
</ul>
<p>Changes made per the advice above are working 80% perfectly.</p>
<p>Now I just have to solve the cases where an “Owner” / “Repository” doesn’t have “Releases” (just “Branches”).</p>
<p>But otherwise very happy :sunglasses:</p>
<p>Cheers, Paul</p>

**Comment by Pamoho** [comment]
<p>G’day team,</p>
<p>After a bit of experimentation I believe I have a solution for those Repositories not organised into Releases.</p>
<p>If I drop the “/releases/latest” off the API “call” I get a JSON page with:</p>
<ul>
<li>created_at</li>
<li>updated_at</li>
<li>pushed_at</li>
</ul>
<p>I need to check that “Updated At” / “Pushed At” are present in every case.  If so this is a good fall-back for not having “Published At”.</p>
<p>But I think we’ve now wrapped up the issue!</p>
<p>Cheers, Paul</p>
